src/test/regress/sql/failure_create_index_concurrently.sql

```diff
@@ -81,6 +81,13 @@ SELECT citus.mitmproxy('conn.allow()');
 SELECT * FROM run_command_on_workers($$SELECT count(*) FROM pg_indexes WHERE indexname LIKE 'idx_index_test%' $$)
 WHERE nodeport = :worker_2_proxy_port;
 
+-- test unique concurrent index creation failure when there are duplicates
+CREATE TABLE index_test_2 (a int, b int);
+SELECT create_distributed_table('index_test_2', 'a');
+INSERT INTO index_test_2 VALUES (1, 1), (1, 2);
+CREATE UNIQUE INDEX CONCURRENTLY index_test_2_a_idx ON index_test_2(a);
+DROP INDEX CONCURRENTLY IF EXISTS index_test_2_a_idx;
```

Comment:
It could be nice to remove the duplicate data and retry creating the index to show that it works.